### PR TITLE
Protect against accidental misuse of XCom.get_value()

### PR DESCRIPTION
**Pull request by potiuk**
The XCom.get_value has been added in 2.3.0 and while there are
cases it should be used in the providers when task are mapped,
in order to keep compatibility with earlier versions of Airlfow,
the XCom.get_value() should only be used when ti_key is not None.

We check for the construct used in community providers automatically
and also add a documentation for users who would like to use
dynamic task mapping featuers in their own providers.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
Updated it so the error looks a bit nicer:

image

**Comment by potiuk** [comment]
I'd love to merge it quickly so that I could incorporate this in the changelogs (hidden entry but avoids future release pollution) to include that one.

File: airflow/providers/amazon/aws/operators/emr.py [hunk 1]
```diff
@@ -244,7 +244,7 @@ def get_link(
         :param dttm: datetime
         :return: url link
         """
-        if ti_key:
+        if ti_key is not None:
```
Comment (ashb) [inline airflow/providers/amazon/aws/operators/emr.py:247]: This makes no difference does it? (since we don't define a custom book on TIKey)

Comment (potiuk) [reply airflow/providers/amazon/aws/operators/emr.py:247]: Nope. But it's easier to standardize as convention. Both ways were used in the code and I arbitrary chose `not None` as more explicit.

Comment (potiuk) [reply airflow/providers/amazon/aws/operators/emr.py:247]: This way we have only one "standard" way to check.

**Review by dstandish — APPROVED** [review]

**Comment by potiuk** [comment]
Rebased to take advantage of the "docs" build fix

**Pull request merged**